src/backend/expungeservice/expunger/analyze.py

```diff
@@ -118,3 +134,166 @@ def __init__(self, name, dob, cases):
         self.name = name
         self.dob = dob
         self.cases = cases
+
+    def num_charges(self):
+        num = 0
+        for case in self.cases:
+            num += case.num_charges()
+        return num
+
+class ResultCode(enum.Enum):
+    INELIGIBLE = 'Ineligible'
+    ELIGIBLE = 'Eligible'
+    FURTHER_ANALYSIS = 'Further analysis needed'
+    EXAMINE = 'Examine'
+    NO_ACTION = 'Take no action'
+    OPEN_CASE = 'Open Case'
+
+"""
+A class for storing individual analysis steps
+
+Attributes:
+    result: A boolean that's the result of the check
+    check: A string that breifly describes the check
+    check_desc: A string that elaborates the check (optional)
+"""
+Check = collections.namedtuple('Check', 'result check check_desc',
+                               defaults=[None])
+
+class Result(object):
+    """
+    A class for storing analysis results
+
+    Attributes:
+        code: A ResultCode instance.
+        statute: A Statute instance that's associated with arrived result.
+        date: A datetime.date instance that specifies eligibility date.
+        checks: A list of Check instances that describes the analysis done.
+    """
+    def __init__(self, code=None, checks=None,
+                 statute=None, date=None):
+        self.code = code
+        self.checks = checks
```

Comment:
Can you add some tests for these checks? If not we can remove this and add it as a feature to do be done at a later time.

Reply:
Do you mean the individual checks? If so, yes it would be nice to have tests that exercise all the paths in the decision tree. I can add tests for the ones that can be added easily and defer other checks.